sdk/core/core-client-rest/src/common.ts

```diff
@@ -48,6 +51,31 @@ export type RequestParameters = {
    * Path parameters for custom the base url
    */
   pathParameters?: Record<string, any>;
+
+  /**
+   * The number of milliseconds a request can take before automatically being terminated.
+   */
+  timeout?: number;
+
+  /**
+   * Callback which fires upon upload progress.
+   */
+  onUploadProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * Callback which fires upon download progress.
+   */
+  onDownloadProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * The signal which can be used to abort requests.
+   */
+  abortSignal?: AbortSignalLike;
+
+  /**
+   * Options used when tracing is enabled.
+   */
+  tracingOptions?: OperationTracingOptions;
```

Comment:
There are other properties in OperationOptions, which I am not sure if I should added into it.
```
/**
   * Options to override serialization/de-serialization behavior.
   */
  serializerOptions?: SerializerOptions;

  /**
   * A function to be called each time a response is received from the server
   * while performing the requested operation.
   * May be called multiple times.
   */
  onResponse?: RawResponseCallback;
```

Reply:
`onResponse` could be useful, but we should definitely not include `serializerOptions` since neither DPG nor RLC use the old serializer functionality.